System/OsPath/Data/ByteString/Short/Word16.hs

```diff
@@ -647,6 +650,23 @@ splitWith p = \(assertEven -> sbs) -> if
               | otherwise -> a : go (tail b)
 
 
+-- | Check whether one string is a substring of another.
+isInfixOf :: ShortByteString -> ShortByteString -> Bool
+isInfixOf sbs = \s -> null sbs || not (null $ snd $ GHC.Exts.inline breakSubstring sbs s)
+
+
+breakSubstring :: ShortByteString -- ^ String to search for
+               -> ShortByteString -- ^ String to search in
+               -> (ShortByteString, ShortByteString) -- ^ Head and tail of string broken at substring
+breakSubstring pat = go mempty
+ where
+    go prefix s = let (h, t)  = GHC.Exts.inline BS.breakSubstring pat s
+                      prefix' = prefix `append` h
+                  in if | even (BS.length prefix') -> (prefix', t)
```

Comment:
I think the length of `prefix` is always even. Consequently the length of `prefix'` is even when the length of `h` is even.

So I guess you could check the length of `h` instead of the length of `prefix'` here, which I think will require less allocation.